endpointmanager/pkg/chplquerier/vendorquerier_test.go

```diff
@@ -0,0 +1,412 @@
+package chplquerier
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/onc-healthit/lantern-back-end/endpointmanager/pkg/endpointmanager"
+	th "github.com/onc-healthit/lantern-back-end/endpointmanager/pkg/testhelper"
+
+	"github.com/spf13/viper"
+)
+
+var testCHPLVendor1 chplVendor = chplVendor{
+	DeveloperID:   448,
+	DeveloperCode: "1447",
+	Name:          "Epic Systems Corporation",
+	Website:       "http://www.epic.com",
+	Address: chplAddress{
+		AddressID: 844,
+		Line1:     "1979 Milky Way",
+		Line2:     nil,
+		City:      "Verona,",
+		State:     "WI",
+		Zipcode:   "53593",
+		Country:   "USA",
+	},
+	LastModifiedDate: "1582575046948",
+	Status: chplStatus{
+		ID:     1,
+		Status: "Active",
+	},
+}
+
+var testCHPLVendor2 chplVendor = chplVendor{
+	DeveloperID:   222,
+	DeveloperCode: "1221",
+	Name:          "Cerner Corporation",
+	Website:       "http://www.cerner.com",
+	Address: chplAddress{
+		AddressID: 873,
+		Line1:     "2800 Rockcreek Parkway",
+		Line2:     nil,
+		City:      "Kansas City,",
+		State:     "MO",
+		Zipcode:   "64117",
+		Country:   "USA",
+	},
+	LastModifiedDate: "1585158101131",
+	Status: chplStatus{
+		ID:     1,
+		Status: "Active",
+	},
+}
+
+var testVendor1 endpointmanager.Vendor = endpointmanager.Vendor{
+	Name:          "Epic Systems Corporation",
+	DeveloperCode: "1447",
+	CHPLID:        448,
+	URL:           "http://www.epic.com",
+	Location: &endpointmanager.Location{
+		Address1: "1979 Milky Way",
+		City:     "Verona,",
+		State:    "WI",
+		ZipCode:  "53593"},
+	Status:             "Active",
+	LastModifiedInCHPL: time.Date(2020, time.February, 24, 20, 10, 46, 0, time.UTC),
+}
+
+func Test_makeVendorURL(t *testing.T) {
+
+	// basic test
+
+	apiKey := viper.GetString("chplapikey")
+	viper.Set("chplapikey", "tmp_api_key")
+	defer viper.Set("chplapikey", apiKey)
+
+	expected := "https://chpl.healthit.gov/rest/developers?api_key=tmp_api_key"
+
+	actualURL, err := makeCHPLVendorURL()
+	th.Assert(t, err == nil, err)
+
+	actual := actualURL.String()
+	th.Assert(t, expected == actual, fmt.Sprintf("Expected %s to equal %s.", actual, expected))
+
+	// test invalid domain and error handling
+
+	chplDomainOrig := chplDomain
+	chplDomain = "http://%41:8080/" // invalid domain
+	defer func() { chplDomain = chplDomainOrig }()
+
+	_, err = makeCHPLVendorURL()
+	switch errors.Cause(err).(type) {
+	case *url.Error:
+		// ok
+	default:
+		t.Fatal("Expected url error")
+	}
+}
+
+func Test_convertVendorJSONToObj(t *testing.T) {
+	var ctx context.Context
+	var err error
+
+	// basic test
+
+	vendorListJSON := `{
+		"developers": [
```

Comment:
Any reason why endpointmanager/pkg/chplquerier/testdata/chpl_vendors.json can't be used for this?

Reply:
mostly to compare against the expected JSON so I don't need to have a much larger comparison list.